### PR TITLE
Location Editor and and some related changes

### PR DESCRIPTION
**Pull request by UncannyValley21**
- Adds a Location Editor
- Extends x and y rotation for RmbBlock3dObjectRecord
- Added missing data on building export
- Fixed issue where overwritten buildings would not get a its name correctly from the generated name seed

### CODE REVIEW

**Comment by UncannyValley21** [comment]
So, as someone who has never made a pull request before, am I just waiting for approval now to have this merged, or do I need to do something more? I have no idea what the procedure is here....

File: Assets/Game/Addons/LocationEditor/Editor/LocationEditor.cs [hunk 1]
```diff
@@ -0,0 +1,537 @@
...
+using UnityEngine;
+using DaggerfallWorkshop.Utility.AssetInjection;
+using System.IO;
+
+/// <summary>
+/// Location editor
+/// </summary>
+namespace DaggerfallWorkshop.Loc {
+
+    public class LocationEditor : EditorWindow {
```
Comment (TheLacus) [inline Assets/Game/Addons/LocationEditor/Editor/LocationEditor.cs]: I would suggest not to use an abbreviation for the namespace. The summary should also be placed above the class definiton, not the namespace.

File: Assets/Game/Addons/LocationEditor/LocationEditorObject.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
+using UnityEngine;
+
+public class LocationEditorObject : MonoBehaviour {
```
Comment (TheLacus) [inline Assets/Game/Addons/LocationEditor/LocationEditorObject.cs]: This class is not inside a namespace.

File: Assets/Game/Addons/UnityConsole/Console/Scripts/DefaultCommands.cs [hunk 3]
```diff
@@ -1,3 +1,25 @@
...
+
+102
+669125Interkarma/daggerfall-unity
+ Code Issues 20 Pull requests 25 Projects 0 Security Insights
+daggerfall-unity/Assets/Game/Addons/UnityConsole/Console/Scripts/DefaultCommands.cs
+@TheLacus TheLacus Add console command 'AddBook'
+b2df12d 2 hours ago
+@Interkarma@ajrb@Allofich@TheLacus@MeteoricDragon@Lypyl@petchema@midopa
+2343 lines (2027 sloc)  110 KB
+
```
Comment (TheLacus) [inline Assets/Game/Addons/UnityConsole/Console/Scripts/DefaultCommands.cs]: What happened here?

File: Assets/Game/Addons/UnityConsole/Console/Scripts/DefaultCommands.cs [hunk 4]
```diff
@@ -26,8 +26,7 @@ public class DefaultCommands : MonoBehaviour
     {
         public static bool showDebugStrings = false;
 
-        void Start()
-        {
+        void Start() {
```
Comment (TheLacus) [inline Assets/Game/Addons/UnityConsole/Console/Scripts/DefaultCommands.cs]: This repository uses the original style.

The entire file was changed and then reverted without apparent reason. Can you try to rebase and cleanup commit history a bit?

**Comment by TheLacus** [comment]
> So, as someone who has never made a pull request before, am I just waiting for approval now to have this merged, or do I need to do something more? I have no idea what the procedure is here....

Interkarma will have to review and merge; there are a lot of additions and changes so i won't take a decision it in his place. 

I think is great that this is being worked on but I believe you should add some additional documentation to explain why specific changes to core are necessary and how the editor backend is organized. As of now reviewing so many lines of code is overwhelming so i understand why this PR has been sitting here more than usual.

**Comment by TheLacus** [comment]
@ajrb Are you ok with these changes?

**Pull request closed without merging**